### ISSUE TITLE
Extensions client page layout and components

### ISSUE THREAD

**Issue by joshuatf** [woocommerce/woocommerce-admin#6138]
Blocked by https://github.com/woocommerce/woocommerce-admin/issues/6134 and https://github.com/woocommerce/woocommerce-admin/issues/6137

This issue will involve creating the extensions page layout and the `<Extension/>` component.  Note that links, favorites, and some other functionality is not required to be finished to begin work on this PR.

* Extensions layout
* Extensions component (consider this in `packages/` for reusability in the WooCommerce ecosystem).

![all-extensions-2](https://user-images.githubusercontent.com/10561050/105397023-8c5a3c00-5bee-11eb-8932-263e2dabae5d.png)

**Comment by joelclimbsthings** [comment]
Closing prior to the merge as it's been on hold for over a year, and there's uncertainty around this project.
